src/common/playlists/PlayThumbnail.jsx

```diff
@@ -42,29 +42,41 @@ const PlayThumbnail = ({ play }) => {
   };
 
   useEffect(() => {
-    // Set the cover image
-    // if it is passed as a meta data
-    if (play.cover) {
-      setCover(play.cover);
-    } else {
-      // if it is not passed as a meta data
-      // check in the play folder for a cover image
-      // with the name cover.png
-      import(`plays/${play.slug}/cover.png`)
-        .then((Cover) => {
-          setCover(Cover.default);
-        })
-        .catch((err) => {
-          // if there is no cover image, set a default image
+    async function setPlayCover() {
+      // Set the cover image
+      // if it is passed as a meta data
+      if (play.cover) {
+        setCover(play.cover);
+      } else {
+        // if it is not passed as a meta data
+        // check in the play folder for a cover image
+        // with the name cover, having the following accepted image formats
+        const acceptedImgExtensions = [`png`, `jpg`, `jpeg`];
+        const imgPromises = [];
+        acceptedImgExtensions.map((ext) =>
+          imgPromises.push(import(`plays/${play.slug}/cover.${ext}`))
+        );
+
+        const response = await Promise.allSettled(imgPromises);
+
+        const fulfilledResult = response.find(
+          (result) => result.status === 'fulfilled' && result.value.default
+        );
```

Comment:
I have one suggestion to make.
Since the highlighted piece of code is repeated within 3 components, we could create a util function for this purpose and use that function across these components.

Reply:
@A-HK Could you please respond to this?

Reply:
Hey ! So sorry, seem to have missed the notification of this review.
Sure, I'll try doing that @supminn @atapas